docs/book/src/plugins/external-plugins.md

```diff
@@ -0,0 +1,157 @@
+# Extending Kubebuilder with external plugins
+
+## Overview
+
+Kubebuilder's functionality can be extended through the use of external plugins.
+
+An external plugin is an executable that implements an execution pattern that Kubebuilder knows how to interact with. Since an external plugin is an executable it can be implemented in any language. 
+
+The Kubebuilder CLI loads the external plugin in the specified path and interacts with it through `stdin` & `stdout`.
+
+## When is it useful?
+
+- If you want to create helpers or addons on top of the scaffolds done by Kubebuilder's default scaffolding.
+
+- If you design customized layouts and want to take advantage of functions from Kubebuilder library.
+
+- If you are looking for implementing plugins in a language other than `Go`.
+
+## How to write it?
+
+The inter-process communication between your external plugin and Kubebuilder is through the standard I/O.
+
+Your external plugin can be written in any language, given it adheres to the [PluginRequest][PluginRequest] and [PluginResponse][PluginResponse] type structures.
+
+`PluginRequest` encompasses all the data Kubebuilder collects from the CLI and previously executed plugins in the plugin chain.
+Kubebuilder conveys the marshaled PluginRequest (a `JSON` object) to the external plugin over `stdin`.
+
+Below is a sample JSON object of the `PluginRequest` type:
```

Comment:
```suggestion
Below is a sample JSON object of the `PluginRequest` type, triggered by "kubebuilder init --plugins v1alpha1 --domain my.domain:
```

Reply:
Thanks for the nice suggestions👍
Just applied!